packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -214,18 +211,18 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
   Widget build(BuildContext context) {
     final bool enabled = widget.enabled;
     final CupertinoThemeData themeData = CupertinoTheme.of(context);
-    final Color primaryColor = themeData.primaryColor;
-    final Color backgroundColor = widget.color == null
+    final Color? primaryColor = themeData.primaryColor;
+    final Color? backgroundColor = widget.color == null
       ? (widget._filled ? primaryColor : null)
       : CupertinoDynamicColor.resolve(widget.color, context);
 
-    final Color foregroundColor = backgroundColor != null
+    final Color? foregroundColor = backgroundColor != null
       ? themeData.primaryContrastingColor
       : enabled
         ? primaryColor
         : CupertinoDynamicColor.resolve(CupertinoColors.placeholderText, context);
 
-    final TextStyle textStyle = themeData.textTheme.textStyle.copyWith(color: foregroundColor);
+    final TextStyle textStyle = themeData.textTheme!.textStyle.copyWith(color: foregroundColor);
```

Comment:
This seems a little strange. What guarantees that `textTheme` is not null?

/cc @HansMuller What's the pattern for dealing with null theme values? Do the widgets usually have a default value in case the theme is also null?

Reply:
TextTheme, which is not a "theme" by our current standards but is too old to change, only has non-null text styles by default. Most (all?) widgets assume that TextTheme properties are non-null

Reply:
the `!` is on the themeData property though. Is `themeData.textTheme` guaranteed to be non-null?

Reply:
Yes, ThemeData.raw(), which is used by the ThemeData factory, requires textTheme to be non-null.

Reply:
Looks like Material's ThemeData is specified that way, but the CupertinoThemeData used here doesn't do that, though.

Reply:
Looks like maybe this getter here should be non-nullable since createDefaults never returns null? https://github.com/flutter/flutter/blob/fb08acf10c63c6eaeb6f32931a965f18cbf28eb9/packages/flutter/lib/src/cupertino/theme.dart#L254-L256

Reply:
FTR: of the 73 ThemeData.raw() parameters, buttonColor, applyElevationOverlayColor, and cupertinoOverrideTheme are _not_ required to be non-null.

https://github.com/flutter/flutter/issues/66060

Reply:
Sorry, it's easy to be misled by a small window into a large file. The `themeData` we're referring to here is a CupertinoThemeData, not ThemeData, and all of its properties can be null because [cupertinoOverrideTheme](https://api.flutter.dev/flutter/material/ThemeData/cupertinoOverrideTheme.html).

Reply:
That's what I was referring in https://discord.com/channels/608014603317936148/684483384839503894/755100965811519627:
> it looks like all properties (primaryColor, primaryContrastingColor, textTheme...) could be non-nullable BUT CupertinoThemeData.noDefault() makes that impossible

Basically material currently relays on the nullability of those properties to work (see https://github.com/flutter/flutter/blob/d26268bb9e6d713a73d6148da7fa75936d442741/packages/flutter/lib/src/material/theme_data.dart#L1764-L1774) by using [_NoDefaultCupertinoThemeData](https://github.com/flutter/flutter/blob/d26268bb9e6d713a73d6148da7fa75936d442741/packages/flutter/lib/src/cupertino/theme.dart) that seems to be the only place where the properties can be null.

I thought it could be possible to refactor the code to make those properties non-nullable but I haven't manage to make it work :-(

So is there something I have to do in this PR regarding this comment thread?

Reply:
Thanks for help!